.github/workflows/doc.yml

```diff
@@ -0,0 +1,38 @@
+name: Documentation
+
+on:
+  push:
+    branches: [master]
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Get version
+        id: get_version
+        run: |
+          echo $(cat VERSION)
+          echo ::set-output name=VERSION::$(cat VERSION)
+
+      - name: Update Doxyfile version
+        env:
+          VERSION: ${{ steps.get_version.outputs.VERSION }}
+        run: sed -i -e "s|SNOWPLOW_TRACKER_VERSION|${VERSION}|g" ./.doxygen/Doxyfile
+
+      - name: Doxygen
+        uses: mattnotmitt/doxygen-action@v1
+        with:
+          working-directory: '.doxygen'
+          doxyfile-path: './Doxyfile'
+    
+      - name: Deploy to GitHub Pages
+        if: success()
+        uses: crazy-max/ghaction-github-pages@v2
```

Comment:
There seem to be some far more popular Actions for deploying pages rather than this one. If this works though then I have no issue with it but thought I'd point it out, as more stars often means better future support.
https://github.com/marketplace/actions/github-pages-action
https://github.com/marketplace/actions/deploy-to-github-pages